### ISSUE TITLE
Why are there separate libraries for axial and cross flow cases?

### ISSUE THREAD

**Issue by hcOnel** [turbinesFoam/turbinesFoam#270]
Intuitively, a cross flow solver would also be capable of solving axial flow cases, right? Why the need of creating different libraries for each?

**Comment by daa-mec** [comment]
What do you mean by different libraries?

**Comment by hcOnel** [comment]
By libraries, I mean dynamic libraries (shared objects), more accurately, the source codes. When running a case, one has to choose     `type            crossFlowTurbineALSource` or `axialFlowTurbineALSource` in `fvOptions`. Cross flow cases are cases in which the flow direction is not coincident with the turbine axis, am I wrong?

**Comment by hcOnel** [comment]
Or are they used for HAWT and VAWT distinction? If so, my bad.

**Comment by daa-mec** [comment]
Yes, I understand that is HAWT and VAWT.

**Comment by hcOnel** [comment]
Ok then, `axialFlowTurbineALSource` is the one to use for HAWTs. Can it handle a yawed wind turbine case?

**Comment by daa-mec** [comment]
From What I have read in the code, I think the correction commented on #268 is done, it should be fine to run HAWTs with let say, inflow velocity with a yaw angle. But I think no body has done any test on that yet. In fact, I am particularly interested in finishing reading the code and do some test cases, including yawed inflow conditions.

**Comment by hcOnel** [comment]
If a rotor deflection is given by changing the rotor axis in fvOptions, as, say:
`        axis                (1 0 0.1);`
would this also change the coordinate system of other components like tower and hub?

**Comment by petebachant** [comment]
There are not different libraries for different turbines--but there are different objects for them. Cross-flow could be vertical or horizontal, but always perpendicular. Axial flow is for a HAWT.

I think both `yaw` and `tilt` should be added to the axial flow turbine model for these properties to avoid needing to set the axis manually. There is some work to be done there for sure.

**Comment by daa-mec** [comment]
Perhaps Pete should know better.

I am not sure, but I think a tilt angle is not supported looking at how the blade geometries are defined. Blades, and its elements, use the vertical direction to set the geometry. From the `verticalDirection_` an the `axis_` is determine the `azimuthalDirection_` (tangential direction). And then the blades are rotated to its azimuthal position (using the `axis_`) and the blade elements pitched using the `spanDirection` (which depends on the vertical direction).

So if you set axis_ with a tilt angle, you would have to change the verticalDirection as well so the blades are constructed appropriately. But then, I think  (haven't check that yet) verticalDirection is used for the tower and the nacelle.

In conclusion, I think the code right as it is, at the moment, does not support a tilt angle.  

Certainly, it would be a good idea to add this functionality. Perhaps it could be done a test, with no tower, nor nacelle, nor hub. And prescribing a tilted axis, and a "verticalDirection" perpendicular to the defined axis, and see what is the result.

**Comment by petebachant** [comment]
When defining the turbine, a free stream direction, yaw, and tilt should be supplied rather than an axis vector. This makes more sense from a usability standpoint.

**Comment by hcOnel** [comment]
> I think both yaw and tilt should be added to the axial flow turbine model for these properties to avoid needing to set the axis manually. There is some work to be done there for sure.

While at it, implementing the capability for a HAWT to face the wind depending on the velocity vector measured at a reasonable point in the mesh (say, the center of rotor for the simplest case) for cases in which the boundary conditions are dynamic, wouldn't be that difficult I assume? Would this kind of implementation need the definition of rotor and/or blade coordinate system in the code to be changed? I think that would require;

- Defining the rotor C.S. with respect to the absolute C.S.,
- Then defining the blade C.S with respect to the rotor C.S.

as the most reasonable approach.
